contracts/voting-verifier/src/execute.rs

```diff
@@ -243,17 +194,18 @@ pub fn end_poll(deps: DepsMut, env: Env, poll_id: PollID) -> Result<Response, Co
     let poll = POLLS
         .may_load(deps.storage, poll_id)?
         .ok_or(ContractError::PollNotFound)?
-        .finish(env.block.height)?;
+        .try_map(|poll| poll.finish(env.block.height).map_err(ContractError::from))?;
 
     POLLS.save(deps.storage, poll_id, &poll)?;
 
-    let poll_result = poll.result();
-
-    match poll {
-        state::Poll::Messages(_) => end_poll_messages(deps, poll_id, &poll_result)?,
-        state::Poll::ConfirmWorkerSet(_) => end_poll_worker_set(deps, poll_id, &poll_result)?,
+    let poll_result = match &poll {
+        Poll::Messages(poll) | Poll::ConfirmWorkerSet(poll) => poll.state(),
     };
 
+    if matches!(poll, state::Poll::ConfirmWorkerSet(_)) {
+        end_poll_worker_set(deps, poll_id, &poll_result)?;
+    }
+
```

Comment:
We could probably do some cleanup here, and delete messages in the poll that didn't achieve quorum. Those messages could just be spam.

Reply:
https://axelarnetwork.atlassian.net/browse/AXE-2650